src/persist/src/indexed/mod.rs

```diff
@@ -1042,6 +1042,16 @@ impl IndexedSnapshot {
     pub fn since(&self) -> Antichain<u64> {
         self.1.since.clone()
     }
+
+    /// Returns the frontier of contained updates.
+    fn upper(&self) -> Antichain<u64> {
+        // TODO(aljoscha): should this be the upper of UnsealedSnapshot
+        // or the TraceSnapshot?
+        //
+        // After writing some data an sealing, UnsealedSnaphot.upper() will be at [], meaning empty
+        // frontier, meaning: we're at +Inf. Which is not what we want.
+        self.1.upper()
```

Comment:
@ruchirK & @danhhz: This is still unclear. I think it's correct two take the upper of the trace, but I'm not 100% sure. On a reader that is just opened for listening, I imagine the unsealed to always be empty and the trace is what matters. Whenever we seal the unsealeds, the listener would also get the `Seal` message.

Reply:
So it turned out I led you astray in my previous review, my apologies! The ts_upper on UnsealedSnapshot is an optional filter used by one internal codepath. We should probably rename that to ts_upper_filter to avoid the same confusion in the future. (ts_lower is also a filter, but this one is always used. if we rename ts_upper, we should probably rename ts_lower as well to match.)

What we actually want here is to explicitly put the sealed timestamp as a field on IndexedSnapshot like we already do for the SeqNo. You can get it in `do_snapshot` by calling `get_seal` on the `Trace` that's already available in that method